state/l1infotree.go

```diff
@@ -19,53 +20,40 @@ type L1InfoTreeLeaf struct {
 type L1InfoTreeExitRootStorageEntry struct {
 	L1InfoTreeLeaf
 	L1InfoTreeRoot  common.Hash
-	L1InfoTreeIndex uint64
+	L1InfoTreeIndex uint
 }
 
-var (
-	// TODO: Put the real hash of Leaf 0, pending of deploying contracts
-	leaf0Hash = [32]byte{} //nolint:gomnd
-)
-
 // Hash returns the hash of the leaf
 func (l *L1InfoTreeLeaf) Hash() common.Hash {
 	timestamp := uint64(l.Timestamp.Unix())
 	return l1infotree.HashLeafData(l.GlobalExitRoot.GlobalExitRoot, l.PreviousBlockHash, timestamp)
 }
 
 // AddL1InfoTreeLeaf adds a new leaf to the L1InfoTree and returns the entry and error
-func (s *State) AddL1InfoTreeLeaf(ctx context.Context, L1InfoTreeLeaf *L1InfoTreeLeaf, dbTx pgx.Tx) (*L1InfoTreeExitRootStorageEntry, error) {
-	allLeaves, err := s.GetAllL1InfoRootEntries(ctx, dbTx)
-	if err != nil {
-		log.Error("error getting all leaves. Error: ", err)
+func (s *State) AddL1InfoTreeLeaf(ctx context.Context, l1InfoTreeLeaf *L1InfoTreeLeaf, dbTx pgx.Tx) (*L1InfoTreeExitRootStorageEntry, error) {
+	var newIndex uint
+	gerIndex, err := s.GetLatestIndex(ctx, dbTx)
+	if err != nil && !errors.Is(err, ErrNotFound) {
+		log.Error("error getting latest l1InfoTree index. Error: ", err)
 		return nil, err
+	} else if err == nil {
+		newIndex = gerIndex + 1
 	}
-	root, err := buildL1InfoTree(allLeaves)
+	log.Debug("latestIndex: ", gerIndex)
+	root, err := s.l1InfoTree.AddLeaf(newIndex, l1InfoTreeLeaf.Hash())
```

Comment:
Instead of looking for next index in DB, doesnt make more sense that the `s.l1InfoTree.AddLeaf`  returns the leafIndex?

Reply:
There is a check in the AddLeaf function to use the right one. If there is a mismatch for some reason an error will be returned